python/lsst/atmospec/utils.py

```diff
@@ -636,7 +636,7 @@ def makeQuery(dataId):
         extraInputCollections = ensure_iterable(extraInputCollections)
         inputs.extend(extraInputCollections)
 
-    butler = dafButler.Butler(repo, writeable=True, collections=inputs)
+    butler = dafButler.Butler(repo, instrument='LATISS', writeable=True, collections=inputs)
```

Comment:
Are you sure this is needed given that the RegistryDefaults setting below overrides this value? Can you put the `instrument='LATISS'` in the RegistryDefaults constructor on line 643? This should allow you to remove it from the butler.get lower down.


Also, we never usually want `logging.basicConfig` in a function since it messes up the global log settings. Are you running this from notebooks that haven't been configured by the log configurator I wrote that sets lsst logs to INFO by default?

Reply:
Yeah, I can probably rejig the `instrument` part. The logging stuff - yes, all the time. We've talked about it countless times but we've never converged on sorting out logging in non-RSP notebooks, and it's always an issue. I totally hear you about `logging.basicConfig` being bad (IIRC `daf_butler`'s use of it caused me headaches too when doing log capture in RA's CI suite), but I've never been able to find a way around getting that working everywhere without resorting to that 😕

Reply:
(1) I'm trying to say that you setting `instrument=` in the Butler constructor isn't doing anything because you immediately reset it in the RegistryDefaults call later (which I think is why you need to add the `instrument=` override later on)
(2) if butler is resetting logging then that's a bug. It should only be doing that via `butler` or `pipetask` command-lines.

Reply:
1) Thanks - fixed.
2) I'll see if I can track it down. I was seriously struggling to find what was causing it, and I _think_ it was that, but I ended up taking a fairly nuclear approach to stopping anything downstream from being able to do that in the end, so I could be mistaken about it being the butler. I remember it happening in a surprising number of places though, and it being _very_ hard to eliminate all of them, hence just using the big hammer in the end.